test/product-import.spec.coffee

```diff
@@ -470,3 +470,137 @@ describe 'ProductImport', ->
       updatedProduct = @import._ensureDefaults(sampleProduct)
       expect(updatedProduct).toEqual expectedProduct
 
+  describe ':: Custom reference resolution', ->
+
+    it ' :: should detect reference type attribute', ->
+      sampleReferenceAttribute =
+        name: 'foobar'
+        value:
+          value: 'some value'
+          resolvePredicate: 'some predicate query'
+          endpoint: 'some endpoint'
```

Comment:
The point is that you are adding extra fields to the attribute. In order to keep it encapsulated, you should namespace them:

```coffeescript
name: 'foobar'
value:
  value: 'some value'
  _custom:
    resolvePredicate: '...'
    endpoint: '...'
    # etc.
```